test/functional/user_sessions_controller_test.rb

```diff
@@ -154,61 +154,74 @@ class UserSessionsControllerTest < ActionController::TestCase
     assert_equal "Successfully logged out.",  flash[:notice]
   end
 
+  test 'sign up and login via provider basic flow for twitter' do
+    assert_not_nil OmniAuth.config.mock_auth[:twitter1]
+    #Omniauth hash is present
+    request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:twitter1]
+    assert_not_nil request.env['omniauth.auth']
+    #Sign Up for a new user
+    post :create
+    assert_equal "You have successfully signed in. Please change your password using the link sent to you via e-mail.",  flash[:notice]
+    #Log Out
+    post :destroy
+    assert_equal "Successfully logged out.",  flash[:notice]
+    #auth hash is present so login via a provider
+    post :create
+    assert_equal "Successfully logged in.",  flash[:notice]
+  end
 
-    test 'sign up and login via provider basic flow for twitter' do
-      assert_not_nil OmniAuth.config.mock_auth[:twitter1]
-      #Omniauth hash is present
-      request.env['omniauth.auth'] =  OmniAuth.config.mock_auth[:twitter1]
-      assert_not_nil request.env['omniauth.auth']
-      #Sign Up for a new user
-      post :create
-      assert_equal "You have successfully signed in. Please change your password using the link sent to you via e-mail.",  flash[:notice]
-      #Log Out
-      post :destroy
-      assert_equal "Successfully logged out.",  flash[:notice]
-      #auth hash is present so login via a provider
+  test 'sign up and login via provider basic flow for twitter user with no email' do
```

Comment:
Noting this is now protected by a specific test for the Twitter-login-with-no-email scenario!